internal/storage/download_test.go

```diff
@@ -12,22 +12,88 @@
 package storage
 
 import (
+	"crypto/x509"
+	"encoding/pem"
+	"io/ioutil"
 	"os"
 	"path/filepath"
+	"reflect"
 	"runtime"
-	"strings"
 	"testing"
 )
 
 const (
-	validCert     = "testdata/valid_cert.pem"
-	validKey      = "testdata/valid_key.pem"
-	expiredCert   = "testdata/expired_cert.pem"
-	expiredKey    = "testdata/expired_key.pem"
-	untrustedCert = "testdata/untrusted_cert.pem"
-	untrustedKey  = "testdata/untrusted_key.pem"
+	validCert      = "testdata/valid_cert.pem"
+	validKey       = "testdata/valid_key.pem"
+	expiredCert    = "testdata/expired_cert.pem"
+	expiredKey     = "testdata/expired_key.pem"
+	untrustedCert  = "testdata/untrusted_cert.pem"
+	untrustedKey   = "testdata/untrusted_key.pem"
+	sslCertFileEnv = "SSL_CERT_FILE"
 )
 
+var sslCertFile string
+
+func isCertAddedToSystemPool(t *testing.T, certFile string) bool {
+	t.Helper()
+
+	certs, err := x509.SystemCertPool()
+	if err != nil {
+		t.Logf("error getting system certificate pool - %v", err)
+		return false
+	}
+	data, err := ioutil.ReadFile(certFile)
+	if err != nil {
+		t.Logf("error reading certificate file %s - %v", certFile, err)
+		return false
+	}
+	block, _ := pem.Decode(data) // ignore rest bytes
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		t.Logf("error parsing certificate %s - %v", certFile, err)
+		return false
+	}
+	subjects := certs.Subjects()
+	for i := 0; i < len(subjects); i++ {
+		if reflect.DeepEqual(subjects[i], cert.RawSubject) {
+			return true
+		}
+	}
+	return false
+}
+
+func setSSLCerts(t *testing.T) {
+	t.Helper()
+
+	// SystemCertPool does not work on Windows: https://github.com/golang/go/issues/16736
+	// Fixed in 1.18
+	if runtime.GOOS != "linux" {
```

Comment:
Actually, this check is very restrictive. Operating systems like FreeBSD, Solaris etc. will be skipped. It's even more complicated because SSL_CERT_FILE doesn't work on macOS. In the end, the test should not run on Windows and macOS.